torch/_subclasses/functional_tensor.py

```diff
@@ -190,23 +191,32 @@ def mark_mutation_hidden_from_autograd(self) -> None:
 
 
 class FunctionalTensorMode(TorchDispatchMode):
-    def __init__(self):
+    def __init__(self, pre_dispatch=False):
         self.is_on_stack = False
         self.enter_stack = []
         # Indicates to our torch_dispatch dispatching infra that
         # this is an "infra" mode with lower dispatching precedence.
         self._mode_key = torch._C._TorchDispatchModeKey.FUNCTIONAL
         # This will be turned off later for pre-dispatch functionalization
         self.decompose_composite_implicit_ops = True
+        self.pre_dispatch = pre_dispatch
```

Comment:
nit: just remove this field, and make any checks below `if self._dispatch_key = torch._C.DispatchKey.PreDispatch`?